main.py

```diff
@@ -137,8 +142,10 @@ def main():
         logging.error("Error retrieving configuration, exiting")
         sys.exit(1)
 
-    modbus_client = setup_modbus_client(configuration)
-    mqtt_client = setup_mqtt_client(configuration)
+    error_handler = setup_error_handler(configuration)
```

Comment:
do you need to pass the entire configuration here? wouldn't you only need a string or simple type instead?

Reply:
I prefer to pass the entire config in all cases, generally in the interest of minimising the code in main.py (which is not covered by unit tests or coverage checks). For error handler there's some conditional logic where the relevant config settings depend on whether the error topic is set or not, and I'd rather do that checking inside error handler itself than in main.py.

Reply:
yeah, I see your point here 👍